### PR TITLE
Fix flaky deployment recovery test

### LINKED ISSUE #10731: MultiPartitionDeploymentRecoveryTest is flaky
**Summary**

- How often does the test fail? once
- Does it block your work? no
- Do we suspect that it is a real failure? no

**Failures**

<details><summary>Example assertion failure</summary>
<pre>
io.camunda.zeebe.test.util.stream.StreamWrapperException: No event found matching the criteria
	at java.base/java.util.Optional.orElseThrow(Optional.java:403)
	at io.camunda.zeebe.test.util.stream.StreamWrapper.getFirst(StreamWrapper.java:77)
	at io.camunda.zeebe.engine.processing.deployment.MultiPartitionDeploymentRecoveryTest.shouldReDistributeAfterRecovery(MultiPartitionDeploymentRecoveryTest.java:60)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:42)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:80)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:72)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:55)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.lambda$execute$1(JUnitPlatformProvider.java:234)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:228)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:175)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:131)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)</pre>
</details>

**Hypotheses**

It could happen that the deployment is already fully distributed before the restart. As a result, the assertion after the restart would fail.

**Logs**

<details><summary>Logs</summary>
<pre>
LOGS
</pre>
</details>

### PR DESCRIPTION
**Pull request by remcowesterhoud**
## Description

The `MultiPartitionDeploymentRecoveryTest` is flaky. After analyzing why this is we came to the conclusion that this test is currently not testing what is intended, which would be verifying that a deployment works after recovering from a failed deployment partition.

In order to fix this flakiness we have deleted this test and replaced with a different integration test. This test verifies that a deployment gets redistributed in the event of a deployment partition restart. In order to achieve this we pause one of the partition of the cluster. This is to make sure we don't process the deployment before we stop the deployment partition.

Once we've verified that the 3rd partition has finished it's deployment as expected we will resume the second partition and restart the deployment partition. After this the test verifies that the deployment is completed as expected.

## Related issues

closes #10731 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/DeploymentClusteredTest.java [hunk 1]
```diff
@@ -102,6 +103,49 @@ public void shouldDistributedDeploymentWhenStoppedBrokerIsRestarted() {
...
 
+  @Test
+  public void shouldRedistributeDeploymentWhenDeploymentPartitionIsRestarted() {
+    // given
+    final var deploymentPartitionLeader =
+        clusteringRule.getLeaderForPartition(Protocol.DEPLOYMENT_PARTITION).getNodeId();
+    final var secondPartitionLeader = clusteringRule.getLeaderForPartition(2).getNodeId();
+    final var adminServiceLeaderTwo =
+        clusteringRule.getBroker(secondPartitionLeader).getBrokerContext().getBrokerAdminService();
+    adminServiceLeaderTwo.pauseStreamProcessing();
```
Comment (saig0) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/DeploymentClusteredTest.java:114]: :wrench: We could add a comment to explain why we pause the partition.

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/DeploymentClusteredTest.java [hunk 2]
```diff
@@ -102,6 +103,49 @@ public void shouldDistributedDeploymentWhenStoppedBrokerIsRestarted() {
...
+
+    assertThat(
+            RecordingExporter.deploymentDistributionRecords()
+                .withIntent(DeploymentDistributionIntent.COMPLETED)
+                .withPartitionId(3)
+                .findFirst())
+        .isPresent();
+
+    clusteringRule.stopBroker(deploymentPartitionLeader);
+    adminServiceLeaderTwo.resumeStreamProcessing();
```
Comment (saig0) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/DeploymentClusteredTest.java:134]: :wrench: We could move this part in the `given` test block. 

The deployment is part of building up the scenario. The `when` block (i.e. the action) is the restart of the broker.

**Review by saig0 — APPROVED** [review]
@remcowesterhoud Looks good. :+1: 

I have some minor suggestions. Please have a look.

**Comment by remcowesterhoud** [comment]
bors merge

**Pull request merged**